### PR TITLE
Modal a11y updates

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
Closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/31444

Specifically addresses these two issues:
1. Upon opening, the id "va-modal-title" is announced and not the heading itself as aria-label is used instead of aria-labelledby
2. The close button has an aria-label "close the warning modal," which can be simplified to just "close"

I'm not sure if this should be a patch or a minor version bump so I went with minor.

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

File: packages/react-components/src/components/Modal/Modal.jsx
Comment (bkjohnson) [inline packages/react-components/src/components/Modal/Modal.jsx:162]: @joshkimux can I get your eyes on the aria changes?

**Comment by erikphansen** [comment]
@bkjohnson This has a failing test in CI. But I cannot seem to run the unit tests locally. Running `yarn test` from the `react-components` dir fails with output like:

```
ERROR in ./src/components/Banner/Banner.jsx
Module not found: Error: Can't resolve '@department-of-veterans-affairs/web-components/react-bindings' in '/Users/erikhansen/adhoc/component-library/packages/react-components/src/components/Banner'
 @ ./src/components/Banner/Banner.jsx 40:0-88 139:42-49
 @ ./src/components/Banner/Banner.unit.spec.jsx
 @ ./src sync \.spec\.js[x]{0,1}$
 @ ./testing/tests.js
ℹ ｢wdm｣: Failed to compile.
01 11 2021 11:47:43.445:INFO [karma-server]: Karma v5.2.3 server started at http://localhost:9876/
01 11 2021 11:47:43.446:INFO [launcher]: Launching browsers ChromeCustom with concurrency unlimited
01 11 2021 11:47:43.512:INFO [launcher]: Starting browser ChromeHeadless
01 11 2021 11:47:45.288:INFO [Chrome Headless 95.0.4638.69 (Mac OS 10.15.7)]: Connected on socket omVSf6IHOAutrhrPAAAA with id 53667501
Chrome Headless 95.0.4638.69 (Mac OS 10.15.7) ERROR
  Uncaught Error: Cannot find module '@department-of-veterans-affairs/web-components/react-bindings'
  at tests.js:228029:171

  Error: Cannot find module '@department-of-veterans-affairs/web-components/react-bindings'
      at webpackMissingModule (tests.js:228029:45)
      at Module../src/components/Banner/Banner.jsx (tests.js:228029:181)
      at __webpack_require__ (tests.js:20:30)
      at Module../src/components/Banner/Banner.unit.spec.jsx (tests.js:228210:69)
      at __webpack_require__ (tests.js:20:30)
      at webpackContext (tests.js:227043:9)
      at Array.forEach (<anonymous>)
      at Module../testing/tests.js (tests.js:238552:21)
      at __webpack_require__ (tests.js:20:30)
      at tests.js:84:18
Chrome Headless 95.0.4638.69 (Mac OS 10.15.7): Executed 0 of 0 ERROR (0.736 secs / 0 secs)
```

**Comment by erikphansen** [comment]
Thanks @bkjohnson! I was missing the web components `build` step. Did I not see the documentation for this or does the documentation just not exist?

**Comment by erikphansen** [comment]
@bkjohnson I had to make a change to get the test to pass so will wait for your 👍 before I merge.

**Review by bkjohnson — APPROVED** [review]
Regarding the documentation - you didn't miss it, it just doesn't exist yet `:homer-disappear:`. The monorepo structure is pretty new, and I was focused on getting that finished fast in order to get through the code freeze, so we are working on adding necessary documentation where it needs to go.

However, we do have CI setup now that will publish the packages anytime a release gets tagged. I have a PR out with documentation for that, but once you merge this you should be able to go ahead and publish:

https://github.com/department-of-veterans-affairs/component-library/pull/210/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R16

**Pull request merged**
